ipfs-pinning-service.yaml

```diff
@@ -45,6 +45,25 @@ response. The old pin object is deleted automatically.
 
 Pin object can be removed via `DELETE /pins/{cid-of-pin-object}`
 
+# Custom metadata
+
+Pinning Services are encouraged to add support for additional features by
+leveraging optional `meta` attributes:
+
+- `PinStatus.meta[progress]` –  % progress of ongoing pinning operation
```

Comment:
I assume the operation that's progressing here is collecting all the blocks that make up a DAG under a given CID?

If so, how does the pinning service know what the progress is?  If it's a dag-pb, fair enough, the links have a Tsize property that may or may not be accurate but what if it's a dag-cbor node that could have any number of CIDs that point to onward blocks?

Or is this just a best-guess estimate?

Reply:
Just best-guess. Probably not the best example, clarified and added better one in